bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Scheduler.java

```diff
@@ -86,6 +86,8 @@
     private ListenerRegistry listenerRegistry;
     private AtomicReference<ItemGroup> objectGroup = new AtomicReference<>();
 
+    private static final int BALLERINA_MAX_STACK_SIZE = 256;
```

Comment:
Shall we call this `BALLERINA_MAX_YIELD_DEPTH`. Max limit for stack size is governed by java, we only limit the stack size of the sleeping thread, ie max depth a yield can happen.